src/Aspire.Hosting/ResourceBuilderExtensions.cs

```diff
@@ -2327,7 +2327,9 @@ public static IResourceBuilder<T> WithVSCodeDebugSupport<T>(this IResourceBuilde
             });
         }
 
-        return builder.WithAnnotation(new SupportsDebuggingAnnotation(projectPath, debugAdapterId, requiredExtensionId));
+        builder.WithAnnotation(new SupportsDebuggingAnnotation(projectPath, debugAdapterId, requiredExtensionId));
+        
+        return builder;
```

Comment:
```suggestion
        return builder.WithAnnotation(new SupportsDebuggingAnnotation(projectPath, debugAdapterId, requiredExtensionId));
```

(nit) this can be reverted.